extensions/csv/src/main/java/io/deephaven/csv/sinks/Sink.java

```diff
@@ -0,0 +1,5 @@
+package io.deephaven.csv.sinks;
+
+public interface Sink<TARRAY> {
+    void write(TARRAY src, int srcOffset, long destOffset, int size);
```

Comment:
Unfortunately it's not always increasing.
Phase 1 starts at some arbitrary midpoint N and then appends until it is done.
Phase 2 starts at 0 and fills up to N

Reply:
Understood. Does it make sense at construction time, or as a new method on Sink, to pass along N?

```java
void init(int totalRows);
```

Reply:
It's possible but there's a performance cost. The algorithm is "streaming" in nature and so it is already writing values to the Sink before it even knows the total size of the input. We *could* implement this but it would mean that we can't start the "column parsing" step until the "input reading" step is completely done, and so this would take away one of our opportunities for parallelism.

Reply:
I wonder if it would be interesting for the Sink to be able to concurrently access the "state" of the writer - essentially, the number of rows the write has written, and if that count is the final count or not.

Reply:
Yeah, there are several things that might be possible and reasonable.
I'd like to understand the use case a little better.
On one hand, telling the sink as little as possible preserves our options for the future.
On the other hand, as you point out, we do have information at certain points that we could share.

For example, our current operations could also be expressed like this

1. Skip N elements (could be zero)
2. Append a chunk to the end of the destination data structure (this repeats an arbitrary number of times)
3. Note I am done. Now start the filling phase at offset 0
4. Fill elements starting at the last offset (this repeats an arbitrary number of times)

We could literally have an interface that conveys these four operations to the Sink if we wanted, if we felt that that gave the Sink some optimization opportunity it didn't otherwise have.

Reply:
Due to your feedback, I have added an "appending" flag to the interface, which makes the logic of the writers a lot simpler. It's not (nearly) everything you had in mind but it's an improvement for now, until we decide on something better.

Reply:
Resolving for now. We can come back to this interface at a future date.